syncmap/syncmap.go

```diff
@@ -0,0 +1,63 @@
+package syncmap
+
+import (
+	"sync"
+)
+
+// Map is a regular map but synchronized with a mutex.
+type Map[K comparable, V any] struct {
+	mu sync.Mutex
+	m  map[K]V
+}
+
+// New returns a new syncmap.
+func New[K comparable, V any]() *Map[K, V] {
+	return &Map[K, V]{
+		m: make(map[K]V),
+	}
+}
+
+// Load returns the value for a key.
+func (m *Map[K, V]) Load(key K) (V, bool) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	v, ok := m.m[key]
+	return v, ok
+}
+
+// Store sets the value for a key.
+func (m *Map[K, V]) Store(key K, value V) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.m[key] = value
+}
+
+// Delete deletes a key.
+func (m *Map[K, V]) Delete(key K) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	delete(m.m, key)
+}
+
+// Len returns the number of elements in the map.
+func (m *Map[K, V]) Len() int {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return len(m.m)
+}
+
+// All iterates over all elements in the map
+func (m *Map[K, V]) All(yield func(K, V) bool) {
```

Comment:
The convention is that All() always returns an iter.Seq/2, rather than being the iterator itself, for consistency.